lib/src/puppetdb-cli.cc

```diff
@@ -116,6 +208,54 @@ size_t write_body(char *ptr, size_t size, size_t nmemb, void *userdata){
     return size * nmemb;
 }
 
+class JsonQueueWrapper {
+  public:
+    typedef char Ch;
+    JsonQueueWrapper(JsonQueue& stream) : stream_(stream) {}
+    Ch Peek() const {  // 1
+        int c = Front();
+        return c == char_traits<char>::eof() ? '\0' : (Ch)c;
+    }
+    Ch Take() {  // 2
+        int c = Get();
+        return c == char_traits<char>::eof() ? '\0' : (Ch)c;
+    }
+    size_t Tell() const { return (size_t)stream_.size(); }  // 3
+    Ch* PutBegin() { assert(false); return 0; }
+    void Put(Ch c) { assert(false); }
+    void Flush() { assert(false); }
+    size_t PutEnd(Ch* c) { assert(false); return 0; }
+  private:
+    int Front() const { return stream_.front(); }
+    int Get() { return stream_.pop(); }
+    JsonQueueWrapper(const JsonQueueWrapper&);
+    JsonQueueWrapper& operator=(const JsonQueueWrapper&);
+    JsonQueue& stream_;
+};
+
+void write_pretty(UserData& userdata) {
+    // This will make sure to block until we have parsed the header
+    userdata.stream.front();
+
+    boost::string_ref content_type{ userdata.content_type };
+    if (content_type.starts_with("application/json")) {
+        rapidjson::Reader reader;
+        JsonQueueWrapper is(userdata.stream);
+        char writeBuffer[65536];
```

Comment:
I'm not going to admit this was copy and paste from their docs... but it might have been copy and paste from the rapidjson docs. How big to we think the buffer should be?